modules/basics/src/main/java/com/opengamma/basics/date/ImmutableHolidayCalendar.java

```diff
@@ -346,6 +346,43 @@ private LocalDate shiftPrev(int baseYear, int baseMonth, int baseDom, int amount
 
   //-------------------------------------------------------------------------
   @Override
+  public LocalDate nextLastOrSameInMonth(LocalDate date) {
+    ArgChecker.notNull(date, "date");
+    try {
+      // day-of-month: no alteration as method is one-based and same is valid
+      return shiftNextLastSame(date.getYear(), date.getMonthValue(), date.getDayOfMonth());
+
+    } catch (ArrayIndexOutOfBoundsException ex) {
+      if (startYear == 0) {
+        return HolidayCalendar.super.next(date);
```

Comment:
Wrong super call? Shouldn't it be ``HolidayCalendar.super.nextLastOrSameInMonth(date);``?